cupy/_core/core.pyx

```diff
@@ -2056,6 +2058,38 @@ cdef class _ndarray_base:
         """
         return dlpack.toDlpack(self)
 
+    def get_ipc_handle(self):
+        """
+        Returns an instance of :class:`IPCHandle`, a serializable class which
+        holds array information and IPC handle for the array. This instance
+        can be passed to other processes (e.g. via multiprocessing).
+
+        On the destination process, use the .get() method to create a new
+        :class:`ndarray` object that shares the allocation from the original process.
+        The IPC handle can be closed manually by calling the .close() method.
+        After that, the destination can no longer use the shared array object.
+        Otherwise, the handle is closed automatically when the instance is deleted
+        using del or goes out of scope.
+
+        Note:
+            The handle cannot be open on the source process or on a destination
+            process created using the 'fork' method.
+
+        Returns:
+            cupy._core.ipc.IPCHandle: Serializable class with
+            Array information and IPC handle.
+
+        Example:
+            >>> # Source process
+            >>> import cupy
+            >>> arr = cupy.ones((3,3))
+            >>> arr_ipc = arr.get_ipc_handle()
+            >>> # Destination process
+            >>> arr = arr_ipc.get()
+        """
+        handle = ipcGetMemHandle(self.data.ptr)
+        return IPCHandle(self.shape, self.size, self.dtype, self.strides, handle)
```

Comment:
```suggestion
        return IPCMemoryHandle(self)
```

I think it's better to retain the reference to `self` in the handle instance. Otherwise `a = cupy.arange(10).get_ipc_handle()` won't work as the backing ndarray will be freed immediately.

https://docs.nvidia.com/cuda/archive/9.0/cuda-runtime-api/group__CUDART__DEVICE.html#group__CUDART__DEVICE_1g01050a29fefde385b1042081ada4cde9

> Calling [cudaFree](https://docs.nvidia.com/cuda/archive/9.0/cuda-runtime-api/group__CUDART__MEMORY.html#group__CUDART__MEMORY_1ga042655cbbf3408f01061652a075e094) on an exported memory region before calling [cudaIpcCloseMemHandle](https://docs.nvidia.com/cuda/archive/9.0/cuda-runtime-api/group__CUDART__DEVICE.html#group__CUDART__DEVICE_1g02bb3632b5d223db6acae5f8744e2c91) in the importing context will result in undefined behavior.

It is also better to document that users are responsible for keeping the IPC instance alive in the source process for the lifetime of the corresponding array in the destination process.